controllers/harborconfiguration_controller.go

```diff
@@ -0,0 +1,396 @@
+	hErr := &harborerrors.ErrProjectNotFound{}
+
+	if err != nil && errors.Is(err, hErr) {
+		err = client.NewProject(ctx, project)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err == nil {
+		// Probably dead code as you cant edit project in UI
+		update := &modelv2.Project{
+			Name:       harborConfiguration.Spec.ProjectReq.ProjectName,
+			RegistryID: srcRegistry.ID,
+		}
+		err = client.UpdateProject(ctx, update, project.StorageLimit)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, err
+}
+
+func (r *HarborConfigurationReconciler) replicationRuleReconciliation(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, registry modelv2.Registry, client *apiv2.RESTClient) (ctrl.Result, error) {
+	srcRegistry, err := client.GetRegistryByName(ctx, harborConfiguration.Spec.Replication.RegistryName)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	reqFilters := make([]*modelv2.ReplicationFilter, 0)
+	for _, v := range harborConfiguration.Spec.Replication.Filters {
+		temp := modelv2.ReplicationFilter{}
+		err := json.Unmarshal(v.Raw, &temp)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		reqFilters = append(reqFilters, &temp)
+	}
+
+	var reqDestinationRegistry *modelv2.Registry
+	if harborConfiguration.Spec.Replication.DestinationRegistry != nil {
+		err = json.Unmarshal(harborConfiguration.Spec.Replication.DestinationRegistry.Raw, reqDestinationRegistry)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+	}
+
+	var reqTrigger *modelv2.ReplicationTrigger
+	if harborConfiguration.Spec.Replication.Trigger != nil {
+		err = json.Unmarshal(harborConfiguration.Spec.Replication.Trigger.Raw, &reqTrigger)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	hErr := &harborerrors.ErrNotFound{}
+	replicationFound, err := client.GetReplicationPolicyByName(ctx, harborConfiguration.Spec.Replication.Name)
+
+	if err != nil && errors.Is(err, hErr) {
+		err = client.NewReplicationPolicy(ctx,
+			reqDestinationRegistry,
+			srcRegistry,
+			harborConfiguration.Spec.Replication.ReplicateDeletion,
+			harborConfiguration.Spec.Replication.Override,
+			harborConfiguration.Spec.Replication.EnablePolicy,
+			reqFilters,
+			reqTrigger,
+			harborConfiguration.Spec.Replication.DestinationNamespace,
+			harborConfiguration.Spec.Replication.Description,
+			harborConfiguration.Spec.Replication.Name)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err == nil {
+		update := modelv2.ReplicationPolicy{
+			Name:          harborConfiguration.Spec.Replication.Name,
+			Description:   harborConfiguration.Spec.Replication.Description,
+			SrcRegistry:   srcRegistry,
+			DestNamespace: harborConfiguration.Spec.Replication.DestinationNamespace,
+			DestRegistry:  reqDestinationRegistry,
+			Filters:       reqFilters,
+			Trigger:       reqTrigger,
+			ID:            replicationFound.ID,
```

Comment:
I am would be quite confused if an update here is possible. Is the ID the same and we only update the other fields?